src/super_gradients/training/metrics/metric_utils.py

```diff
@@ -39,6 +39,40 @@ def calc_batch_prediction_detection_metrics_per_class(metrics, dataset_interface
     return results_tuple
 
 
+def calc_batch_prediction_detection_metrics_per_classV2(matching, dataset_interface, iou_thres, silent_mode,
+                                                      per_class_verbosity, class_names, test_loss):
+    images_counter = len(matching)
```

Comment:
Is there any case that len(matching) is 0 ?
If not, we need to remove that if statement.
If there is, then line 72: results_tuple = (mean_precision, mean_recall, mean_ap, mean_f1, *test_loss.average)

might crash.

Reply:
same, this function was removed because outdated and never used: f5182088bd85175c86d345622c1242a924c989e7